src/test/java/org/jabref/gui/fieldeditors/LinkedFileViewModelTest.java

```diff
@@ -170,4 +168,28 @@ void downloadDoesNotOverwriteFileTypeExtension() throws MalformedURLException {
         task.onFailure(Assertions::fail);
         new CurrentThreadTaskExecutor().execute(task);
     }
+
+    @Test
+    void isNotSamePath() {
+        linkedFile = new LinkedFile("desc", tempFile, "pdf");
+        databaseContext = mock(BibDatabaseContext.class);
+        when(filePreferences.getFileNamePattern()).thenReturn("[citationkey]"); // use this variant, as we cannot mock the linkedFileHandler cause it's initialized inside the viewModel
```

Comment:
If it's hard to write tests, then it's a good indication that some refactoring should be done. Indeed, we use the mvvm pattern to make testing easy. Could you thus please add a constructor in the viewmodel that accepts a `linkedFileHandler` .

Reply:
Actually, it's just the comment which is obsolete here, as the linkedFilesHandler is not required and can be tested separately (and already is). Would need the same mocking, so no difference.